Documentation/ABI/JSON.md

```diff
@@ -0,0 +1,227 @@
+# JSON schema
+
+<!--
+This source file is part of the Swift.org open source project
+
+Copyright (c) 2024 Apple Inc. and the Swift project authors
+Licensed under Apache License v2.0 with Runtime Library Exception
+
+See https://swift.org/LICENSE.txt for license information
+See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+-->
+
+This document outlines the JSON schemas used by the testing library for its ABI
+entry point and for the `--experimental-event-stream-output` command-line
+argument. For more information about the ABI entry point, see the documentation
+for [ABI.EntryPoint_v0](https://github.com/search?q=repo%3Aapple%2Fswift-testing%EntryPoint_v0&type=code).
+
+> [!WARNING]
+> This JSON schema is still being developed and is subject to any and all
+> changes including removal from the package.
+
+## Modified Backus-Naur form
+
+This schema is expressed using a modified Backus-Naur syntax. `{`, `}`, `:`, and
+`,` represent their corresponding JSON tokens. `\n` represents an ASCII newline
+character.
+
+The order of keys in JSON objects is not normative. Whitespace in this schema is
+not normative; it is present to help the reader understand the content of the
+various JSON objects in the schema. The event stream is output using the JSON
+Lines format and does not include newline characters (except **one** at the end
+of the `<output-record-line>` rule.)
+
+Trailing commas in JSON objects and arrays are only to be included where
+syntactically valid.
+
+### Common data types
+
+`<string>` and `<number>` are defined as in JSON. `<array:T>` represents an
+array (also defined as in JSON) whose elements all follow rule `<T>`.
+
+```
+<bool> ::= true | false ; as in JSON
+
+<source-location> ::= {
+  "fileID": <string>, ; the Swift file ID of the file
+  "line": <number>,
+  "column": <number>,
+}
+
+<timestamp> ::= {
```

Comment:
Is this `<timestamp>` used elsewhere in the spec?

Reply:
Should be `<instant>` now.